### ISSUE TITLE
Rendering larger csv files in Pivot Table Enhancement

### ISSUE THREAD

**Issue by alexnovac** [nicolaskruchten/pivottable#373]
Hi Nicolas, 

First I want to say Pivottable.js is fantastic!! 

So I ran into issues rendering csv files when the data being used reached 30MB+ or around 82,000 x 39; the browser would simply lock up and consume an ungodly amount of memory. In firefox it would eventually end up loading, in chrome however it would crash every single time.

So at first I said to myself hmmm maybe its just a browser limitation! 

Then I decided to sniff further and realized that the issue with memory usage was the parsing being done by jquery-csv.

Browsing online I found another parsing library by the name of papa parse: 
https://github.com/mholt/PapaParse

Using Papa Parse instead of jquery-csv the memory consumption was around 30% - 40% of what jquery-csv was consuming. The library has a lot of options and natively allows the streaming of a large csv (this was used for example 3) and overall parses the data at a much faster rate.

Code: 

$.get("mps.csv", function(mps) {
                var input = Papa.parse(mps, {skipEmptyLines: true, config: {
            step: function(results,parser) {
                row++;
                $('#pp').html(row);
            }
        } });    
					
				$("#output").pivotUI( input.data, {
                            renderers: $.extend(
                            $.pivotUtilities.renderers, 
                            $.pivotUtilities.gchart_renderers,
							$.pivotUtilities.c3_renderers,
                            $.pivotUtilities.d3_renderers,
							$.pivotUtilities.export_renderers
                            )
                        });
                    });
				});	

Example 1 file with 82k rows using Papa Parse (notice memory consumption):
![dynpivgif1](https://cloud.githubusercontent.com/assets/4553243/9282142/1eb08aee-427f-11e5-832f-661a6767b8ad.gif)

Example 2 file with 82k rows using jquery-csv:
![dynpivgif2bad](https://cloud.githubusercontent.com/assets/4553243/9282151/2e1cfbf2-427f-11e5-81d0-2cbbb67a4d1a.gif)

Example 3 file with 247k rows using Papa Parse:
![dynpivgif3big](https://cloud.githubusercontent.com/assets/4553243/9282174/571817bc-427f-11e5-811c-a3b199e5b999.gif)

**Comment by nicolaskruchten** [comment]
Cool! I'll take a look at PapaParse and maybe upgrade the examples :) Thanks!

**Comment by slester87** [comment]
Using pivottable with large data, I was running into a problem where my browser would crash, ostensibly on the toArrays csv load of a very large .csv file (>150mb). I saw that most of the time was spent on the var input = ... toArrays call and I found this solution. With papaparse, the load times are manageable and the load of the pivottable is no longer crashing. Thanks for pointing this out, it was relatively painless to use.
